torch/csrc/jit/codegen/cuda/index_compute.cpp

```diff
@@ -3026,25 +2799,10 @@ std::vector<RootPredicateInfo> Index::getReferenceRootPredicates(
       start_magic_zero_info = stop_magic_zero_info;
     }
 
-    start_index = start_magic_zero_info.index;
-    stop_index = stop_magic_zero_info.index;
-
-    // Update the loop-index map with the magic-zero protection info
-    // before passing it to the hoisting function
-    std::tie(start_index, stop_index) = hoistPredicates(
-        start_index,
-        stop_index,
-        loops,
-        stop_indexing_from_idgraph.resolved_loop_domains,
-        updateInitialLoopIndexMap(
-            start_indexing_from_idgraph.initial_concrete_index_map,
-            start_magic_zero_info),
-        updateInitialLoopIndexMap(
-            stop_indexing_from_idgraph.initial_concrete_index_map,
-            stop_magic_zero_info),
-        unswitch_or_vec_loop,
-        contig_id,
-        consumer_tv);
+    start_index = GpuLower::current()->commonIndexMap().hoistIndex(
+        start_magic_zero_info.index, loops);
+    stop_index = GpuLower::current()->commonIndexMap().hoistIndex(
+        stop_magic_zero_info.index, loops);
```

Comment:
Instead of hoisting start and stop index, we should be able to just hoist the entire predicate as a boolean expression. I will not do it in this PR, will investigate later.